### PR TITLE
Improve the output of the swagger API for watch events

### PR DESCRIPTION
**Pull request by smarterclayton**
Stopgap to improve this prior to converting watch resources to
versioned objects.

Spawned from #6182

### CODE REVIEW

**Comment by bgrant0607** [comment]
LGTM. Thanks. @thockin can rebase on top of this and regenerate the swagger.

**Comment by brendandburns** [comment]
Tests are failing:
```
# github.com/GoogleCloudPlatform/kubernetes/pkg/watch/json
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/watch/json/decoder_test.go:45: undefined: watchEvent
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/watch/json [build failed]
```

**Comment by bgrant0607** [comment]
Merged multi-port services. Please run hack/update-swagger-spec.sh.

**Comment by smarterclayton** [comment]
Rebased, test fixed, and docs updated.

----- Original Message -----

File: api/swagger-spec/v1beta3.json [hunk 1]
```diff
@@ -7276,38 +7483,10 @@
...
-      "type": "string",
-      "description": "the port to access on the pods targeted by the service; defaults to the service port"
-     }
-    }
-   },
    "v1beta3.ServiceSpec": {
     "id": "v1beta3.ServiceSpec",
     "required": [
-     "ports",
+     "port",
```
Comment (bgrant0607) [inline api/swagger-spec/v1beta3.json]: Something's not right here. This is reverting multi-port services, among other things.

Comment (smarterclayton) [inline api/swagger-spec/v1beta3.json]: Bad merge, let me delete and regen.

----- Original Message -----

Comment (smarterclayton) [inline api/swagger-spec/v1beta3.json]: Fixed, sorry.

----- Original Message -----

**Comment by bgrant0607** [comment]
Weird. I'm not seeing the comments I received via email, nor the latest push.

**Comment by bgrant0607** [comment]
@smarterclayton Maybe try to make clean, rebuild, and regenerate the swagger?

@nikhiljindal Any guess about what's going on with the swagger generation? A stale go-restful dependency?

**Comment by nikhiljindal** [comment]
@smarterclayton : Did you rebuild the binary before running update-swagger-spec?
That script uses the existing binary rather then re-building it.

**Comment by bgrant0607** [comment]
@nikhiljindal He picked up some recent API changes, so I'd guess he did build it.

**Comment by nikhiljindal** [comment]
The type changes and addition of a path parameter in swagger spec are coming from #6143 and #6117.

**Comment by bgrant0607** [comment]
Ok, sorry. Swagger has been updated from earlier changes. Please rebase and re-update swagger again. Feel free to self-merge.

**Comment by bgrant0607** [comment]
@smarterclayton Could we try to get this in before something breaks it again? :-) It might be easier if you didn't update the swagger in this PR.

**Comment by smarterclayton** [comment]
Rebased, split swagger out.

**Comment by bgrant0607** [comment]
LGTM. Thanks.

When looking at this I noticed that the /log content types are wrong.

The shippable failure is a known kubelet test problem.

**Pull request merged**